### ISSUE TITLE
Performance issue with stablelib

### ISSUE THREAD

**Issue by tuyennhv** [ChainSafe/js-libp2p-noise#223]
**Description**
- The js version of chacha20poly1305 takes significant time in lodestar

<img width="1426" alt="Screen Shot 2022-10-08 at 17 29 10" src="https://user-images.githubusercontent.com/10568965/194703050-ab323a25-6a71-486f-97d6-89fffaf55c4a.png">

Need to find a different implementation:
- Try to find a wasm implementation
- Or find a native implementation and do async call there
[1008_feat2_1.2.0_rc.0.cpuprofile.zip](https://github.com/ChainSafe/js-libp2p-noise/files/9738985/1008_feat2_1.2.0_rc.0.cpuprofile.zip)

**Comment by dapplion** [comment]
- @tuyennhv Opened a PR https://github.com/ChainSafe/js-libp2p-noise/pull/224 to add metrics to noise, relevant to this issue is to measure packet rate inbound and outbound so we can understand how expensive noise is relative to benchmarks

**Comment by mpetrunic** [comment]
I don't think it makes sense to track it in this repository as this library allows you to add your own cryptography libraries
